polkadot/runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -345,7 +364,7 @@ impl<T: Config> Pallet<T> {
 		log::debug!(target: LOG_TARGET, "Time weight before filter: {}, candidates + bitfields: {}, disputes: {}", weight_before_filtering.ref_time(), candidates_weight.ref_time() + bitfields_weight.ref_time(), disputes_weight.ref_time());
 
 		let current_session = shared::CurrentSessionIndex::<T>::get();
-		let expected_bits = scheduler::AvailabilityCores::<T>::get().len();
+		let expected_bits = scheduler::ValidatorGroups::<T>::get().len();
```

Comment:
Ugly. (Was before already.) This should be a proper function call. The core number exposed by the scheduler should likely include virtual cores, then we can use that call.

From the outside of the scheduler, actual vs virtual cores should not be different. Virtual cores will simply never have assignments.

Reply:
Accessing storage of another pallet directly should be a big Nogo. Provide an API that exposes what should be exposed and then use that.

Reply:
true. done!